java/grpc-client/src/test/java/com/youtube/vitess/client/grpc/GrpcClientTlsTest.java

```diff
@@ -0,0 +1,203 @@
+package com.youtube.vitess.client.grpc;
+
+import com.google.common.io.Files;
+import com.youtube.vitess.client.Context;
+import com.youtube.vitess.client.RpcClientTest;
+import org.apache.commons.io.IOUtils;
+import org.joda.time.Duration;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import java.io.File;
+import java.io.StringWriter;
+import java.net.InetSocketAddress;
+import java.net.ServerSocket;
+import java.nio.file.Paths;
+import java.util.Arrays;
+
+/**
+ * This tests GrpcClient with a mock vtgate server (go/cmd/vtgateclienttest), over an SSL connection.
+ *
+ * The SSL setup is adapted from "test/encrypted_transport.py"
+ */
+@RunWith(JUnit4.class)
+public class GrpcClientTlsTest extends RpcClientTest {
+
+    private static File certDirectory;
+    private static Process vtgateclienttest;
+    private static int port;
+
+    @BeforeClass
+    public static void setUpBeforeClass() throws Exception {
+        certDirectory = Files.createTempDir();
+        System.out.println("Using cert directory: " + certDirectory.getCanonicalPath());
+
+        createCA();
+        createSignedCert("ca", "03", "vtgate-server", "vtgate server CA");
+        createSignedCert("ca", "04", "vtgate-client", "vtgate client CA");
+
```

Comment:
I'm starting to get a bit over my head in this area, and could use some high-level guidance from @harshit-gangal or @alainjobart.  I'm adapting logic from Alain's `encrypted_transport.py` script, but barely understand much of what I'm adapting.  

I THINK that I just want to:

1. Create a CA
2. Use that CA to create signed certificates for the VTGate server and the client
3. Start a VTGate server process (or `vtgateclienttest` mock in this case), configured with the server cert
4. Start a gRPC client connection, using the client cert

However, the Python script is creating numerous other certs (e.g. separate ones for VTGate and VTTablet)... and even seems to be using separate CA's for some reason.

Can you please give me a sanity check, to be sure I'm not going too far down the wrong path?  I think I just need to better understand what that Python script is doing at the high-level, so I can either replicate it appropriately or else only borrow a subset if it's overkill for this case.  Thanks!